charts/common/templates/_require.tpl

```diff
@@ -1,24 +1,26 @@
 {{/*
-  Checks specifically for a CRD api version and resource.
+  Validate the existence of a Kubernetes API/CRD object and version.
 
-  Use like `{{ include "require.api" ( list "errorMessage.flux" "helm.toolkit.fluxcd.io" "HelmRelease" $) }}`
+  Use like `{{ include "skyfjell.common.require.api" ( list "errorMessage.flux" "helm.toolkit.fluxcd.io" "HelmRelease" $ ) }}`
 */}}
-{{- define "require.api" }}
-  {{- $global :=  ( last . ) }}
-  {{- $tpl := (first . ) }}
-  {{- $api := ( first ( rest . ) ) }}
-  {{- $kind := ( last ( initial . ) ) }}  
+{{- define "skyfjell.common.require.api" }}
+  {{- $ :=  last . }}
+  {{- $apiMessage := first . }}
+  {{- $api := rest . | first }}
+  {{- $kind := initial . | last }}
   {{- $flag := dict "flag" "false" }}
-  {{- range $installed := $global.Capabilities.APIVersions }}
+  {{- $helpMessage := "To disable these checks see: https://github.com/skyfjell/charts/tree/main/charts/common/README.md#api-validation" -}}
+  {{- range $installed := $.Capabilities.APIVersions }}
     {{- $matchApi := (hasPrefix ( printf "%s/" $api ) $installed ) }}
     {{- $matchKind := (hasSuffix ( printf "/%s" $kind ) $installed )}}
-    {{-  if (and $matchApi $matchKind) }}
+    {{-  if and $matchApi $matchKind }}
       {{- $_ := set $flag "flag" "true" }}
     {{- end }}
   {{- end }}
-  {{- if or (not (hasKey $global.Values "crdCheck")) (get $global.Values "crdCheck") -}}
```

Comment:
Are we going to integrate this `.Values.skyfjell.validate.api` into each chart might want to check for its existance? I feel like this might be too nested?